pilot/pkg/config/kube/gateway/controller.go

```diff
@@ -145,124 +341,42 @@ func (c *Controller) Get(typ config.GroupVersionKind, name, namespace string) *c
 }
 
 func (c *Controller) List(typ config.GroupVersionKind, namespace string) []config.Config {
-	if typ != gvk.Gateway && typ != gvk.VirtualService {
-		return nil
-	}
-
-	c.stateMu.RLock()
-	defer c.stateMu.RUnlock()
 	switch typ {
 	case gvk.Gateway:
-		return filterNamespace(c.state.Gateway, namespace)
+		res := slices.MapFilter(c.outputs.Gateways.List(), func(g Gateway) *config.Config {
+			if g.Valid {
+				return g.Config
+			}
+			return nil
+		})
+		return res
 	case gvk.VirtualService:
-		return filterNamespace(c.state.VirtualService, namespace)
+		return slices.Map(c.outputs.VirtualServices.List(), func(e *config.Config) config.Config {
+			return *e
```

Comment:
Gonna keep reading, but can't HTTPRoutes also be invalid? OR do we just not store it